src/Workers/WorkerCommands.php

```diff
@@ -26,6 +29,8 @@ public function __construct(Application $console, Client $client, WorkerProvider
         $this->client = $client;
         $this->workerProvider = $workerProvider;
         $this->outputInterfaceAware = $outputInterfaceAware;
+        
+        $this->messageAdapterFactory = null;
```

Comment:
The point is the missing nullable in type hinting (as we are support php 5, we have to use default value to manage it ... or sadly drop the type hinting)

You're right with the BodyFactory point but in conception view I wanted to preserve some layers of abstraction. That's why we have 2 levels of factories.

For the mandatory part, I've considered that users that don't need to customize theses parts must have nothing to do at all (so no BC break in API). That's why dependencies are optional and that's why there are fallbacks to set standards implementations if they don't provide any